src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/Wedge.cpp

```diff
@@ -150,128 +427,195 @@ std::array<DataVector, 3> Wedge::gradient(
+    if (get_element(result, i) + eps_ < 0.0 or
+        get_element(result, i) - eps_ > 1.0) {
       ERROR(
           "The Wedge transition map was called with coordinates outside "
-          "the set inner and outer surfaces. The inner radius and sphericity "
-          "are (r="
-          << inner_surface_.radius << ",s=" << inner_surface_.sphericity
-          << ") and the outer radius and sphericity are (r="
-          << outer_surface_.radius << ",s=" << outer_surface_.sphericity
-          << "). The inner distance is " << get_element(inner_distance, i)
-          << ", the outer distance is " << get_element(outer_distance, i)
-          << ". The requested point has radius: " << get_element(mag, i));
+          "the set inner and outer surfaces.\nThe requested (centered) point "
+          "is "
+          << source_coords << "\nThe requested (centered) point has radius "
+          << get_element(centered_coords_magnitude, i)
+          << "\nThe inner surface has center, "
+             "radius, and sphericity (c="
+          << inner_surface_.center << ",r=" << inner_surface_.radius
+          << ",s=" << inner_surface_.sphericity
+          << ")\nThe outer surface has center, radius, and sphericity (c="
+          << outer_surface_.center << ",r=" << outer_surface_.radius
+          << ",s=" << outer_surface_.sphericity
+          << ")\nThe distance to the inner surface is "
+          << get_element(inner_distance, i)
+          << "\nThe distance to the outer surface is "
+          << get_element(outer_distance, i));
     }
   }
-#endif  // SPECTRE_DEBUG
+#endif
 }
 
+namespace {
+// struct to allow unpacking version 0 of this class
+struct LegacySurface {
+  double radius{};
+  double sphericity{};
+
+  void pup(PUP::er& p) {
+    p | radius;
+    p | sphericity;
+  }
+};
+}  // namespace
+
 void Wedge::pup(PUP::er& p) {
   ShapeMapTransitionFunction::pup(p);
-  size_t version = 0;
+  size_t version = 1;
   p | version;
   // Remember to increment the version number when making changes to this
   // function. Retain support for unpacking data written by previous versions
   // whenever possible. See `Domain` docs for details.
-  if (version >= 0) {
+  if (version >= 1) {
     p | inner_surface_;
     p | outer_surface_;
+    p | projection_center_;
     p | axis_;
+  } else if (p.isUnpacking()) {
+    LegacySurface inner_surface{};
+    LegacySurface outer_surface{};
+    size_t axis = 0;
+    p | inner_surface;
+    p | outer_surface;
+    p | axis;
+
+    // We don't know the centers of the objects so we just set them to 0
+    inner_surface_ = Surface{
+        {0.0, 0.0, 0.0},
+        inner_surface.radius,
+        inner_surface.sphericity,
+    };
+    outer_surface_ = Surface{
+        {0.0, 0.0, 0.0},
+        outer_surface.radius,
+        outer_surface.sphericity,
+    };
+    // Unfortunately we can't recover the sign for the axis so we just set it
```

Comment:
Why not? This is worrisome since then the serialized and de-serialized objects are not equal. We generally assume that serialization and deserialization is equivalent to a copy and that the two objects should be equal...

Reply:
Because the previous version of this class assumed there was zero offset, I only stored whether the wedge was x,y,z, but not the +/- part because the +/- of the same axis were identical. However, now we need to know +/- for this version. Not really sure what to do about this